### ISSUE TITLE
Method dc.run_gsva error

### ISSUE THREAD

**Issue by chuiqin** [saezlab/decoupler-py#90]
Thanks for developing a nice tool, but I'm getting an error when running `dc.run_gsva`. 
**Describe your question**
You can repeat my mistake with the following simple code.
```
import os
import pandas as pd 
import numpy as np
import scanpy as sc
import decoupler as dc
import argparse
adata = sc.datasets.pbmc3k()
progeny = dc.get_progeny(organism='human', top=500)
dc.run_gsva(
    mat=adata,
    net=progeny,
    source='source',
    target='target',
    verbose=True,
    use_raw=False
) 
```
Then, error occurs:
```
16104 features of mat are empty, they will be removed.
Running gsva on mat with 2700 samples and 16634 targets for 14 sources.
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/home/data/t030408/.local/share/r-miniconda/envs/irGSEA/lib/python3.8/site-packages/decoupler/method_gsva.py", line 229, in run_gsva
    estimate = gsva(m, net, kcdf=kcdf, verbose=verbose)
  File "/home/data/t030408/.local/share/r-miniconda/envs/irGSEA/lib/python3.8/site-packages/decoupler/method_gsva.py", line 145, in gsva
    mat = density(mat, kcdf=kcdf)
  File "/home/data/t030408/.local/share/r-miniconda/envs/irGSEA/lib/python3.8/site-packages/decoupler/method_gsva.py", line 79, in density
    mat = mat_ecdf(mat)
  File "/home/data/t030408/.local/share/r-miniconda/envs/irGSEA/lib/python3.8/site-packages/numba/core/dispatcher.py", line 703, in _explain_matching_error
    raise TypeError(msg)
TypeError: No matching definition for argument type(s) pyobject

```
The version of decoupler-py is 1.5.1. Looking forward to hearing from you.

**Comment by chuiqin** [comment]
With redundant steps, you can get normal results, but it's not elegant
```
import os
import pandas as pd 
import numpy as np
import scanpy as sc
import decoupler as dc
import argparse
adata = sc.datasets.pbmc3k()
progeny = dc.get_progeny(organism='human', top=500)
data = pd.DataFrame.sparse.from_spmatrix(adata.X)
data.index=adata.obs.index
data.columns=adata.var.index
dc.run_gsva(
    mat=data,
    net=progeny,
    source='source',
    target='target',
    verbose=True,
    use_raw=False
) 

```

```
16104 features of mat are empty, they will be removed.
Running gsva on mat with 2700 samples and 16634 targets for 14 sources.
100%|██████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████| 14/14 [00:01<00:00, 12.81it/s]
source            Androgen      EGFR  Estrogen   Hypoxia  JAK-STAT      MAPK      NFkB      PI3K      TGFb      TNFa     Trail      VEGF       WNT       p53
AAACATACAACCAC-1 -0.225219 -0.312801 -0.158113 -0.439287 -0.425172 -0.382872 -0.346634 -0.305657  0.136413 -0.226733 -0.144486 -0.003223 -0.139282 -0.075963
AAACATTGAGCTAC-1 -0.241491 -0.313816 -0.160495 -0.459620 -0.408327 -0.340274 -0.357467 -0.293657  0.104100 -0.242376 -0.081498 -0.018299 -0.097449 -0.112773
AAACATTGATCAGC-1 -0.243690 -0.319014 -0.163602 -0.427772 -0.405724 -0.369693 -0.346125 -0.328842  0.124220 -0.196960 -0.142529 -0.051860 -0.113524 -0.052231
AAACCGTGCTTCCG-1 -0.241322 -0.341083 -0.203071 -0.428081 -0.397036 -0.400800 -0.365822 -0.317726  0.151434 -0.260649 -0.118109 -0.011944 -0.127641 -0.122889
AAACCGTGTATGCG-1 -0.231033 -0.310178 -0.172712 -0.442210 -0.408406 -0.376968 -0.369128 -0.322793  0.133948 -0.250277 -0.158445  0.046179 -0.112514 -0.098578
...                    ...       ...       ...       ...       ...       ...       ...       ...       ...       ...       ...       ...       ...       ...
TTTCGAACTCTCAT-1 -0.258960 -0.321570 -0.183678 -0.434005 -0.376872 -0.358093 -0.392485 -0.320614  0.145620 -0.255973 -0.124077 -0.049329 -0.081492 -0.110295
TTTCTACTGAGGCA-1 -0.197516 -0.295954 -0.177901 -0.422605 -0.389149 -0.391009 -0.379190 -0.336082  0.093761 -0.263410 -0.117101 -0.002173 -0.088386 -0.072928
TTTCTACTTCCTCG-1 -0.230401 -0.303952 -0.160490 -0.454308 -0.424450 -0.391683 -0.377900 -0.285185  0.131662 -0.267680 -0.132266  0.033401 -0.105755 -0.088640
TTTGCATGAGAGGC-1 -0.242077 -0.327503 -0.174127 -0.462896 -0.410184 -0.398138 -0.364154 -0.326809  0.146318 -0.249656 -0.131071  0.023161 -0.100577 -0.085703
TTTGCATGCCTCAC-1 -0.247598 -0.329842 -0.193083 -0.444678 -0.409504 -0.381029 -0.392035 -0.319902  0.152952 -0.268557 -0.134378  0.001907 -0.103010 -0.111178

[2700 rows x 14 columns]

```

**Comment by PauBadiaM** [comment]
Hi @chuiqin 
Thanks for reporting this, it seems I introduced an error in this commit a8b778e10ef21a2a62d4acca18d3fc8a4c469a7e. Now it should be solved in 5e947dffef50486cbde2476d4c41a58bbba04e3e. You can install the latest version from GitHub and it should work:
```
pip install --upgrade git+https://github.com/saezlab/decoupler-py.git
```
Hope this is helpful! Let me know how it goes.
